### PR TITLE
Update kustomize to 3.8.10

### PR DESCRIPTION
**Pull request by nickgerace**
Relevant issue: https://github.com/rancher/fleet/issues/287

### CODE REVIEW

**Comment by nickgerace** [comment]
Need to look at this locally for a bit...

**Comment by nickgerace** [comment]
Slightly changed dependencies based on: https://github.com/kubernetes-sigs/kustomize/blob/kustomize/v3.8.10/kustomize/go.mod

File: pkg/apis/go.mod [hunk 1]
```diff
@@ -1,6 +1,13 @@
 module github.com/rancher/fleet/pkg/apis
 
-go 1.14
+go 1.15
```
Comment (nickgerace) [inline pkg/apis/go.mod:3]: `go 1.15` upgrade to match root `go.mod`

File: go.mod [hunk 2]
```diff
@@ -29,6 +29,10 @@ replace (
 	k8s.io/metrics => k8s.io/metrics v0.20.2
 	k8s.io/mount-utils => k8s.io/mount-utils v0.20.2
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.2
+	sigs.k8s.io/kustomize/api => sigs.k8s.io/kustomize/api v0.6.9
```
Comment (StrongMonkey) [inline go.mod:32]: This needs to be pinned in `require`. Replace is actuallt a hack for k8s because they don't pin their dependency in `go.mod`

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**